test/jasmine/tests/scattermapbox_test.js

```diff
@@ -365,6 +365,23 @@ describe('scattermapbox convert', function() {
         expect(actualText).toEqual(['A', 'B', 'C', 'F', '']);
     });
 
+    it('for lines traces with trailing gaps', function() {
+        var opts = _convert(Lib.extendFlat({}, base, {
+            mode: 'lines',
+            lon: [10, '20', 30, 20, null, 20, 10, null],
+            lat: [20, 20, '10', null, 10, 10, 20, null]
+        }));
+
+        assertVisibility(opts, ['none', 'visible', 'none', 'none']);
+
+        var lineCoords = [
+            [[10, 20], [20, 20], [30, 10]],
+            [[20, 10], [10, 20]]
+        ];
```

Comment:
> I assume it will also handle [10, '20', 30, 20, null, 20, 10, null, null] without problems?

Yes it does: https://github.com/plotly/plotly.js/pull/1421/commits/1a22bebd1776b2d70658ef3c4101655c3839912e